### ISSUE TITLE
[Bug]: avx instruction inside libopenvino_intel_npu_plugin.so leads to crash on system without AVX instruction

### ISSUE THREAD

**Issue by LeptonWu** [openvinotoolkit/openvino#26715]
### OpenVINO Version

2024.4.0

### Operating System

Other (Please specify in description)

### Device used for inference

CPU

### Framework

None

### Issue description

By following 
https://www.intel.com/content/www/us/en/developer/tools/openvino-toolkit/download.html?PACKAGE=OPENVINO_BASE&VERSION=v_2024_4_0&OP_SYSTEM=LINUX&DISTRIBUTION=PIP

install openvino 2024.4.0 in virtual env. then run this test, it crashed with illegal instruction on my pentium g4600 CPU.
dmegs shows that it's from libopenvino_intel_npu_plugin.so.

```
traps: pt_main_thread[34059] trap invalid opcode ip:7f5863245f57 sp:7ffd467020c0 error:0 in libopenvino_intel_npu_plugin.so[7f586322b000+1c1000]
```

My question is: from the plugin name it looks like it's for NPU and why it runs avx instruction when trying init it?  I
don't even have NPU.  If it's from cpu_plugin.so, I can somehow understand the point. Maybe we don't need to compile with avx optimazation for npu plugin? at least for the init part?

BTW, I just manually deleted the libopenvino_intel_npu_plugin.so from the disk and then the same code runs fine and can show there are 2 devices, one CPU, one GPU.

```python
import openvino as ov

core = ov.Core()
for device in core.available_devices:
  print(device)
```

### Issue submission checklist

- [X] I'm reporting an issue. It's not a question.
- [X] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [X] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by kennorsdb** [comment]
I confirm I had the same problem. The workaround works as well.

**Comment by RyanMetcalfeInt8** [comment]
Just to point out here, that many users of our OpenVINO-based Audacity plugins are regularly reporting this issue:

https://forum.audacityteam.org/t/how-to-fix-unable-to-load-the-mod-openvino-module-error/116996/

https://github.com/intel/openvino-plugins-ai-audacity/issues/295

https://forum.audacityteam.org/t/cant-install-openvino-correctly-need-help-with-installation/118089/2

(and many more)

**Comment by RyanMetcalfeInt8** [comment]
By the way, this issue is also present in 2024.3.

**Comment by ilya-lavrenov** [comment]
@dmatveev it's a regression after https://github.com/openvinotoolkit/openvino/pull/24989/files#diff-fab2995529c22d419387ce6a5a141c41cb4195f5e617f50f7281a7819d52966c
Could you please fix it?

Proper solution is to check in runtime that AVX2 is available and run function compiled with AVX2 in runtime.
You can use CPU's approach https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/CMakeLists.txt#L282-L288 to compile the same function for multiple instruction sets and accompanied runtime dispatching.

**Comment by dmatveev** [comment]
@ilya-lavrenov please see CVS-147126, it seems the issue is not related to stepping into ours hand-written AVX2 code (as it is not used during the loading), but is related to AVX2 found in `std` structures which may have it was the plugin was compiled with AVX2.

**Comment by avitial** [comment]
Ref. 154085

**Comment by avitial** [comment]
This issue has been addressed on the recent release 2024.5, please give it a try on your end. Closing for now, feel free to reopen and ask additional questions if the issue persists.
